lib/devise-security/models/password_expirable.rb

```diff
@@ -2,68 +2,101 @@
+    # Adjust the +password_changed_at+ field so that it will require a
+    # password update and save the record (without validations)
+    # @return [Boolean]
+    def need_change_password!
+      return unless password_expiration_enabled?
+      need_change_password
+      save(validate: false)
+    end
+    alias expire_password! need_change_password!
+    alias request_password_change! need_change_password!
+
+    # Clear the +password_changed_at+ field so that it will require a
+    # password update.
+    # @return [void]
+    def need_change_password
+      return unless password_expiration_enabled?
+      self.password_changed_at = nil
+    end
+    alias expire_password need_change_password
+    alias request_password_change need_change_password
+
+    # Set this value to a number of seconds to have passwords expire after a time
+    # Set to +true+ if you want to be able to manually expire passwords on demand
+    # without a time limit.
+    # @return [Numeric, Boolean]
+    def expire_password_after
+      self.class.expire_password_after
+    end
 
-        self.password_changed_at
-      end
+    # When +password_changed_at+ is set to null in the database
+    # the user is required to change their password.  This only happens
+    # on demand or when the column is first added to the table.
+    # @return [Boolean]
+    def password_change_requested?
+      return false unless password_expiration_enabled?
+      return false if new_record?
+      password_changed_at.nil?
+    end
 
-      def expire_password_after
-        self.class.expire_password_after
-      end
+    # Is this password older than the configured expiration timeout
+    # @return [Boolean]
+    def password_too_old?
+      return false if new_record?
+      return false unless password_expiration_enabled?
+      return false if expire_password_on_demand?
+      password_changed_at < expire_password_after.seconds.ago
+    end
+    alias password_expired? password_too_old?
 
-      private
+    private
 
-        # is password changed then update password_cahanged_at
-        def update_password_changed
-          self.password_changed_at = Time.now if (self.new_record? || self.encrypted_password_changed?) && !self.password_changed_at_changed?
-        end
+    # is password changed then update password_changed_at
+    # @note called as a +before_save+ hook
+    def update_password_changed
+      return unless (new_record? || encrypted_password_changed?) && !password_changed_at_changed?
+      self.password_changed_at = Time.zone.now
+    end
 
-        def expired_password_after_numeric?
-          return @_numeric if defined?(@_numeric)
-          @_numeric ||= self.expire_password_after.is_a?(1.class) ||
-            self.expire_password_after.is_a?(Float)
-        end
+    # Enabled if configuration +expire_password_after+ is set to an {Integer}, {Float}, or {true}
+    def password_expiration_enabled?
+      expire_password_after.present? &&
```

Comment:
`` expire_password_after.present?` is already part of `expire_password_on_demand?`